contrib/opencensus-ext-azure/opencensus/ext/azure/utils/__init__.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import locale
+import os
+import platform
+import sys
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+from opencensus.common.version import __version__ as opencensus_version
+from opencensus.common.utils import timestamp_to_microseconds
+from opencensus.ext.azure.protocol import Object
+from opencensus.ext.azure.version import __version__ as extension_version
+
+azure_monitor_context = {
+    'ai.cloud.role': os.path.basename(sys.argv[0]) or 'Python Application',
+    'ai.cloud.roleInstance': platform.node(),
+    'ai.device.id': platform.node(),
+    'ai.device.locale': locale.getdefaultlocale()[0],
+    'ai.device.osVersion': platform.version(),
+    'ai.device.type': 'Other',
+    'ai.internal.sdkVersion': 'py{}:oc{}:ext{}'.format(
+        platform.python_version(),
+        opencensus_version,
+        extension_version,
+    ),
+}
+
+
+def microseconds_to_duration(microseconds):
+    n = (microseconds + 500) // 1000  # duration in milliseconds
+    ms = n % 1000  # millisecond
+    n = n // 1000
+    s = n % 60  # second
+    n = n // 60
+    m = n % 60  # minute
+    n = n // 60
+    h = n % 24  # hour
+    d = n // 24  # day
+    return '{:d}.{:02d}:{:02d}:{:02d}.{:03d}'.format(d, h, m, s, ms)
+
+
+def timestamp_to_duration(start_time, end_time):
+    start_time_us = timestamp_to_microseconds(start_time)
+    end_time_us = timestamp_to_microseconds(end_time)
+    duration_us = int(end_time_us - start_time_us)
+    return microseconds_to_duration(duration_us)
```

Comment:
curious if there is a built in library in python which can be used instead of this custom code? If not should this be a generic utility method part of a utility class rather than being in specific exporter?

Reply:
The "duration" format here is specific to Microsoft Azure Monitor.